From 6cfddf5ba79a7fee1f2e3d6e934461943e01b5ce Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 24 Oct 2016 10:15:37 +0800
Subject: [PATCH] MLK-13379: ASoC: wm8960: workaround no sound issue in master
 mode

The input MCLK is 12.288MHz, The desired output sysclk is 11.2896MHz
the sample rate is 44100Hz, with the pllprescale=2, postscale=sysclkdiv=1,
some chip may have wrong bclk and lrclk output in master mode. then there
will be no sound.
With the pllprescale=1, postscale=2, the output clock is correct. so use
this configuration to workaround this issue.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/codecs/wm8960.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/wm8960.c b/sound/soc/codecs/wm8960.c
index 28eb426..b77b7a3 100644
--- a/sound/soc/codecs/wm8960.c
+++ b/sound/soc/codecs/wm8960.c
@@ -642,7 +642,7 @@ static int wm8960_configure_clocking(struct snd_soc_codec *codec)
 
 	/* check if the sysclk frequency is available. */
 	for (i = 0; i < ARRAY_SIZE(sysclk_divs); ++i) {
-		if (sysclk_divs[i] == -1)
+		if (sysclk_divs[i] != 2)
 			continue;
 		sysclk = freq_out / sysclk_divs[i];
 		for (j = 0; j < ARRAY_SIZE(dac_divs); ++j) {
@@ -666,7 +666,7 @@ static int wm8960_configure_clocking(struct snd_soc_codec *codec)
 	}
 	/* get a available pll out frequency and set pll */
 	for (i = 0; i < ARRAY_SIZE(sysclk_divs); ++i) {
-		if (sysclk_divs[i] == -1)
+		if (sysclk_divs[i] != 2)
 			continue;
 		for (j = 0; j < ARRAY_SIZE(dac_divs); ++j) {
 			sysclk = lrclk * dac_divs[j];
-- 
1.9.1

